tests/integration/test_charm.py

```diff
@@ -102,3 +103,116 @@ async def test_health_checks(ops_test: OpsTest, lightkube_client: Client) -> Non
 
     response = requests.get(base_url, timeout=300, verify=False)
     assert response.status_code == 403
+
+
+@pytest.mark.abort_on_fail
+async def test_auth_proxy_relation(ops_test: OpsTest) -> None:
+    """Ensure that oauth2-proxy is able to provide auth-proxy relation."""
+    requirer_charm_path = Path(f"tests/integration/{AUTH_PROXY_REQUIRER}").absolute()
+    requirer_charm = await ops_test.build_charm(requirer_charm_path, verbosity="debug")
+
+    await ops_test.model.deploy(
+        application_name=AUTH_PROXY_REQUIRER,
+        entity_url=requirer_charm,
+        resources={"oci-image": "kennethreitz/httpbin"},
+        series="jammy",
+        trust=True,
+    )
+
+    await ops_test.model.integrate(f"{AUTH_PROXY_REQUIRER}:ingress", TRAEFIK)
+    await ops_test.model.integrate(f"{AUTH_PROXY_REQUIRER}:auth-proxy", APP_NAME)
+
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME, AUTH_PROXY_REQUIRER, TRAEFIK],
+        status="active",
+        raise_on_blocked=False,
+        timeout=1000,
+    )
+
+
+@pytest.mark.abort_on_fail
+async def test_forward_auth_relation(ops_test: OpsTest) -> None:
+    """Ensure that oauth2-proxy is able to provide forward-auth relation."""
+    await ops_test.model.applications[TRAEFIK].set_config({
+        "enable_experimental_forward_auth": "True"
+    })
+    await ops_test.model.integrate(f"{TRAEFIK}:experimental-forward-auth", APP_NAME)
+
+    await ops_test.model.wait_for_idle([TRAEFIK, APP_NAME], status="active", timeout=1000)
+
+
+@retry(
```

Comment:
question: why do we need the retry here?

Reply:
for an edge case in which the config is not yet populated, this could happen if the oauth2 proxy reconfigures when a `relation-changed` is emitted